packages/block-editor/src/components/link-control/use-remote-url-data.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+
+function useRemoteUrlData( url ) {
+	const [ richData, setRichData ] = useState( null );
+	const [ isFetching, setIsFetching ] = useState( false );
+
+	const { fetchRemoteUrlData } = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+		return {
+			fetchRemoteUrlData: getSettings().__experimentalFetchRemoteUrlData,
+		};
+	}, [] );
+
+	function abortableFetch( theUrl, signal ) {
```

Comment:
Don't need to create this function since `apiFetch` (`fetchRemoteUrlData`) should be able to accept the `signal` property from `AbortController`.

```js
fetchRemoteUrlData(url, { signal })
```